src/hackingBuddyGPT/utils/db_storage/db_storage.py

```diff
@@ -11,10 +11,12 @@ def __init__(self, connection_string: str = parameter(desc="sqlite3 database con
     def init(self):
         self.connect()
         self.setup_db()
-
+    
     def connect(self):
-        self.db = sqlite3.connect(self.connection_string)
+#        self.db = sqlite3.connect(self.connection_string, timeout=10)  # Set timeout to 10 seconds
+        self.db = sqlite3.connect(self.connection_string, check_same_thread=False, timeout=10)
```

Comment:
this is weird. I am trying to understand the error case. Are you running two instances of hackingBuddyGPT in parallel? This would not be supported by the sqlite3 database (that we are currently using). I think the clean solution would rather be to use a separate sqlite database per run (using the `log_db` parameter) or using a database such as postgres/mysql that supports concurrent access